src/components/PopupWithForm.js

```diff
@@ -16,9 +17,13 @@ class PopupWithForm extends Popup {
     return formValues;
   }
 
-  submit(e) {
+  submit(e) { // handles all submitting stuff
     e.preventDefault();
+    const buttonLabel = this._popupElement.querySelector(".modal__save").innerText;
+    this._popupElement.querySelector(".modal__save").innerText = "Saving....";
     this._handleFormSubmit(this._getInputValues());
+    this._popupElement.querySelector(".modal__save").innerText = buttonLabel;
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.

And this way you'll wait till the request is fullfilled